src/main/java/com/jhuguet/sb_taskv1/app/controller/UserController.java

```diff
@@ -59,58 +65,70 @@ public EntityModel<Page<User>> getAll(@RequestParam(defaultValue = "0") int page
             InvalidInputInformation, WrongSortOrder {
         pageResponse.validateInput(page, size);
         Page<User> users = userService.getAll(pageResponse.giveDynamicPageable(page, size, sort));
+        logger.info("Retrieving all users");
 
         return EntityModel.of(users, linkTo(methodOn(UserController.class).getAll(page, size, sort)).withSelfRel());
     }
 
     /**
      * Will retrieve User from Database
      *
-     * @param id ID to look for in Database
      * @return will return pertaining User retrieved from DB
      * @throws IdNotFound Exception thrown when given ID is not found
      */
-    @GetMapping("/{id}")
-    public User get(@PathVariable int id) throws IdNotFound {
+    @GetMapping("/current")
+    public User get(@RequestHeader Map<String, String> headers) throws IOException, BaseException {
+        String jwt = utils.retrieveJwt(headers);
```

Comment:
data about user can be taken from Principal

Reply:
What do you mean should be taken from the principal? What is the principal?